services/oauth/v2/stats.go

```diff
@@ -18,40 +19,57 @@ type OAuthStats struct {
 	isCallToCpApi   bool   // is a call being made to control-plane APIs
 	authErrCategory string // for action=refresh_token -> REFRESH_TOKEN, for action=fetch_token -> "", for action=auth_status_inactive -> auth_status_inactive
 	destDefName     string
-	isTokenFetch    bool              // This stats field is used to identify if a request to get token is arising from processor
 	flowType        common.RudderFlow // delivery, delete
 	action          string            // refresh_token, fetch_token, auth_status_inactive
 }
 
-func (s *OAuthStats) SendTimerStats(startTime time.Time) {
-	statsTags := stats.Tags{
-		"id":              s.id,
-		"workspaceId":     s.workspaceID,
-		"rudderCategory":  s.rudderCategory,
-		"isCallToCpApi":   strconv.FormatBool(s.isCallToCpApi),
-		"authErrCategory": s.authErrCategory,
-		"destType":        s.destDefName,
-		"flowType":        string(s.flowType),
-		"action":          s.action,
+type OAuthStatsHandler struct {
+	stats       stats.Stats
+	defaultTags stats.Tags
+}
+
+func GetDefaultTagsFromOAuthStats(oauthStats *OAuthStats) stats.Tags {
+	return stats.Tags{
+		"id":              oauthStats.id,
+		"workspaceId":     oauthStats.workspaceID,
+		"rudderCategory":  "destination",
+		"isCallToCpApi":   strconv.FormatBool(oauthStats.isCallToCpApi),
+		"authErrCategory": oauthStats.authErrCategory,
+		"destType":        oauthStats.destDefName,
+		"flowType":        string(oauthStats.flowType),
+		"action":          oauthStats.action,
 		"oauthVersion":    "v2",
 	}
-	s.stats.NewTaggedStat(s.statName, stats.TimerType, statsTags).SendTiming(time.Since(startTime))
 }
 
-// SendCountStat Send count type stats related to OAuth(Destination)
-func (s *OAuthStats) SendCountStat() {
-	statsTags := stats.Tags{
-		"oauthVersion":    "v2",
-		"id":              s.id,
-		"workspaceId":     s.workspaceID,
-		"rudderCategory":  s.rudderCategory,
-		"errorMessage":    s.errorMessage,
-		"isCallToCpApi":   strconv.FormatBool(s.isCallToCpApi),
-		"authErrCategory": s.authErrCategory,
-		"destType":        s.destDefName,
-		"isTokenFetch":    strconv.FormatBool(s.isTokenFetch),
-		"flowType":        string(s.flowType),
-		"action":          s.action,
+func NewStatsHandlerFromOAuthStats(oauthStats *OAuthStats) OAuthStatsHandler {
+	defaultTags := GetDefaultTagsFromOAuthStats(oauthStats)
+	return OAuthStatsHandler{
+		stats:       oauthStats.stats,
+		defaultTags: defaultTags,
+	}
+}
+
+func (m *OAuthStatsHandler) mergeTags(tags stats.Tags) stats.Tags {
+	allTags := m.defaultTags
+	for key, value := range tags {
+		allTags[key] = value
 	}
-	s.stats.NewTaggedStat(s.statName, stats.CountType, statsTags).Increment()
+	return allTags
+}
```

Comment:
This should be the other way around. If we don't want to put values into `m.defaultTags`.
```
func (m *OAuthStatsHandler) mergeTags(tags stats.Tags) stats.Tags {
	for key, value := range m.defaultTags {
		tags[key] = value
	}
	return tags
}
```
Add a map lookup check if you don't want to override with default.

Reply:
Used a different way to solve this problem. Please check and let me know if it works